signaler/signaler.go

```diff
@@ -6,20 +7,17 @@ import (
 	"syscall"
 )
 
-var s = make(chan os.Signal, 1)
-
-func init() {
-	sigs := []os.Signal{
-		os.Interrupt,
-		os.Kill,
-		syscall.SIGTERM,
-		syscall.SIGABRT,
-	}
-	signal.Notify(s, sigs...)
+// WaitForInterrupt blocks until a termination signal is received and returns it.
+// It registers a temporary channel, unregisters it via signal.Stop() and returns the received signal.
```

Comment:
🚧 This comment is about how the function behaves internally, which isn't information callers need to know.
```suggestion
```